packages/dds/tree/src/test/simple-tree/objectNode.spec.ts

```diff
@@ -234,10 +236,99 @@ describeHydration(
 				}) {}
 				const n = init(HasId, {});
 				assert.throws(() => {
-					// TODO: AB:9129: this should not compile
+					// @ts-expect-error this should not compile
```

Comment:
This is the test which was tracking the known part of this issue. It now works as expected for non-recursive objects.